waiter/src/waiter/state.clj

```diff
@@ -46,6 +53,37 @@
   (and (zero? slots-used)
        (not-any? #(contains? status-tags %) [:killed :locked])))
 
+(defn- find-max
+  "Uses the provided comparison function to find the max element in collection."
+  [comparison-fn collection]
+  (reduce #(if (pos? (comparison-fn %1 %2)) %1 %2) (first collection) (rest collection)))
+
+(defn- only-lingering-requests?
```

Comment:
Is `only-lingering-request` an extension to the definition of what's `killable?`?  Would that be a more natural place to put this logic?